### DISCUSSION TITLE
How to require another extension in existing extension?

### DISCUSSION THREAD

**Discussion by nikmace** [microsoft/vscode-discussions#424] [Extension Development QnA] [answered]
Hello,

We want to do the following, we have an **extension** that _depends on another extension_. When the user installs our extension it throws an **error** with a message that the required extension is not installed. 

How can we **install the dependent extension programmatically** if it's not yet installed? 
Or how can we **show a user-friendly message** saying the user should install another extension before installing ours? 
Or are the **other solutions** to this?

Thanks in advance!

**Comment by alefragnani** [comment, marked as answer]
Hi,

If your extension _**depends on another extension**_, the proposed approach is to use `extensionDependencies` entry in `package.json`. More details https://code.visualstudio.com/api/references/extension-manifest. 

Hope this helps
